transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -801,6 +801,10 @@ private void deregister(final ChannelPromise promise, final boolean fireChannelI
                 return;
             }
 
+            if (recvHandle != null) {
```

Comment:
I would also check on `fireChannelInactive` because in theory it is possible to deregister a channel without closing it.